numpy/core/src/npymath/npy_math_private.h

```diff
@@ -484,79 +483,4 @@ do {                                                            \
 
 #endif /* !HAVE_LDOUBLE_DOUBLE_DOUBLE_* */
 
-/*
- * Those unions are used to convert a pointer of npy_cdouble to native C99
- * complex or our own complex type independently on whether C99 complex
- * support is available
- */
-#ifdef NPY_USE_C99_COMPLEX
-
-/*
- * Microsoft C defines _MSC_VER
- * Intel compiler does not use MSVC complex types, but defines _MSC_VER by
- * default.
- * since c++17 msvc is no longer support them.
- */
-#if !defined(__cplusplus) && defined(_MSC_VER) && !defined(__INTEL_COMPILER)
-typedef union {
-        npy_cdouble npy_z;
-        _Dcomplex c99_z;
-} __npy_cdouble_to_c99_cast;
-
-typedef union {
-        npy_cfloat npy_z;
-        _Fcomplex c99_z;
-} __npy_cfloat_to_c99_cast;
-
-typedef union {
-        npy_clongdouble npy_z;
-        _Lcomplex c99_z;
-} __npy_clongdouble_to_c99_cast;
-#else /* !_MSC_VER */
-typedef union {
-        npy_cdouble npy_z;
-#ifdef __cplusplus
-        std::complex<double> c99_z;
-#else
-        complex double c99_z;
-#endif
-} __npy_cdouble_to_c99_cast;
-
-typedef union {
-        npy_cfloat npy_z;
-#ifdef __cplusplus
-        std::complex<float> c99_z;
-#else
-        complex float c99_z;
-#endif
-} __npy_cfloat_to_c99_cast;
-
-typedef union {
-        npy_clongdouble npy_z;
-#ifdef __cplusplus
-        std::complex<long double> c99_z;
-#else
-        complex long double c99_z;
-#endif
-} __npy_clongdouble_to_c99_cast;
-#endif /* !_MSC_VER */
-
-#else /* !NPY_USE_C99_COMPLEX */
-typedef union {
-        npy_cdouble npy_z;
-        npy_cdouble c99_z;
-} __npy_cdouble_to_c99_cast;
-
-typedef union {
-        npy_cfloat npy_z;
-        npy_cfloat c99_z;
-} __npy_cfloat_to_c99_cast;
-
-typedef union {
-        npy_clongdouble npy_z;
-        npy_clongdouble c99_z;
-} __npy_clongdouble_to_c99_cast;
-#endif /* !NPY_USE_C99_COMPLEX */
-
-
```

Comment:
This would be a good place to put the new code from `numpy/core/include/numpy/npy_math.h`